solara/server/reload.py

```diff
@@ -152,25 +161,46 @@ def _on_change(self, name):
     def close(self):
         self.watcher.close()
 
+    def get_reload_module_names(self):
+        if self.aggresive_reload:
+            # not sure why, but if we reload pandas, the integration/reload_test.py fails
+            return {
+                k for k in set(sys.modules) - set(self.ignore_modules) if not (k.startswith("solara.server") or k.startswith("anyio") or k.startswith("pandas"))
+            }
+        else:
+            reload = []
+            for name in sorted(sys.modules):
+                mod = sys.modules[name]
+                if name.startswith("solara.server"):
+                    continue  # this will break everything
+                if name in self.ignore_modules:
+                    continue  # nothing we imported from solara itself
+                if hasattr(mod, "__file__") and mod.__file__:
+                    if not mod.__file__.startswith(str(self.root_path)):
```

Comment:
if `mod.getattr("__file__", "MISSING").startswith(str(self.root_path))` would be cleaner IMO

Reply:
thanks, I agree